### PR TITLE
docs: add docs for spreadsheet custom components

### PR DESCRIPTION
**Pull request by ugur-vaadin**
## Description

Add documentation for Spreadsheet Custom Components integration. 

The text content is from [V8 Spreadsheet Charts documentation](https://vaadin.com/docs/v8/spreadsheet/spreadsheet-charts-integration). A minimal example with a button is introduced.

Part of https://github.com/vaadin/platform/issues/3568

image

### CODE REVIEW

File: src/main/java/com/vaadin/demo/component/spreadsheet/SpreadsheetComponents.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                                                            int rowIndex, int columnIndex,
+                                                            Spreadsheet spreadsheet, Sheet sheet) {
+                        return null;
+                    }
+
+                    @Override
+                    public void onCustomEditorDisplayed(Cell cell, int rowIndex,
+                                                        int columnIndex, Spreadsheet spreadsheet,
+                                                        Sheet sheet, Component customEditor) {
+                    }
```
Comment (tomivirkki) [inline src/main/java/com/vaadin/demo/component/spreadsheet/SpreadsheetComponents.java:47]: Would be helpful to also have these methods implemented and explained

Comment (ugur-vaadin) [reply src/main/java/com/vaadin/demo/component/spreadsheet/SpreadsheetComponents.java:47]: Then I suppose we should add another example for an editor. IIRC, these methods are meant to be implemented only if the component is an editor.

Comment (tomivirkki) [reply src/main/java/com/vaadin/demo/component/spreadsheet/SpreadsheetComponents.java:47]: It should be possible to have components and editors in the same example (different cells of course)

Comment (ugur-vaadin) [reply src/main/java/com/vaadin/demo/component/spreadsheet/SpreadsheetComponents.java:47]: That is correct, I will update the example.

File: src/main/java/com/vaadin/demo/component/spreadsheet/SpreadsheetComponents.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
+package com.vaadin.demo.component.spreadsheet;
+
+import com.vaadin.demo.DemoExporter; // hidden-source-line
+import com.vaadin.flow.router.Route;
+import com.vaadin.flow.component.button.Button;
+import com.vaadin.flow.component.Component;
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.Span;
```
Comment (tomivirkki) [inline src/main/java/com/vaadin/demo/component/spreadsheet/SpreadsheetComponents.java:8]: Unused import

File: src/main/java/com/vaadin/demo/component/spreadsheet/SpreadsheetComponents.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                        return customComponent;
+                    }
+
+                    @Override
+                    public Component getCustomEditorForCell(Cell cell,
+                            int rowIndex, int columnIndex,
+                            Spreadsheet spreadsheet, Sheet sheet) {
+                        if (rowIndex != 2 || columnIndex != 2) {
+                            return null;
+                        }
```
Comment (tomivirkki) [inline src/main/java/com/vaadin/demo/component/spreadsheet/SpreadsheetComponents.java:53]: nit: A guard statement with the negative checks doesn't scale well if you want to have more than 1 component in your sheet, which is probably a more typical scenario.

File: src/main/java/com/vaadin/demo/component/spreadsheet/SpreadsheetComponents.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        spreadsheet.createCell(1, 2, "Text Field");
+
+        // tag::snippet[]
+        spreadsheet.setSpreadsheetComponentFactory(
+                new SpreadsheetComponentFactory() {
+
+                    @Override
+                    public Component getCustomComponentForCell(Cell cell,
+                            int rowIndex, int columnIndex,
+                            Spreadsheet spreadsheet, Sheet sheet) {
```
Comment (tomivirkki) [inline src/main/java/com/vaadin/demo/component/spreadsheet/SpreadsheetComponents.java:37]: Could read the value of `sheet` when deciding whether to add the components. Currently, the components are added on every sheet, while the texts are only on the first one:

https://user-images.githubusercontent.com/1222264/219662562-7d09444b-d693-4067-981a-aa29d24e7fce.mp4

File: src/main/java/com/vaadin/demo/component/spreadsheet/SpreadsheetComponents.java (hunk 3 above)
Comment (ugur-vaadin) [reply src/main/java/com/vaadin/demo/component/spreadsheet/SpreadsheetComponents.java]: Addressed in the latest commit

File: src/main/java/com/vaadin/demo/component/spreadsheet/SpreadsheetComponents.java (hunk 2 above)
Comment (ugur-vaadin) [reply src/main/java/com/vaadin/demo/component/spreadsheet/SpreadsheetComponents.java]: Addressed in the latest commit

File: src/main/java/com/vaadin/demo/component/spreadsheet/SpreadsheetComponents.java (hunk 4 above)
Comment (ugur-vaadin) [reply src/main/java/com/vaadin/demo/component/spreadsheet/SpreadsheetComponents.java:36]: Addressed in the latest commit

**Review by tomivirkki — APPROVED** [review]

**Pull request merged**